src/qlever/commands/index.py

```diff
@@ -38,12 +39,95 @@ def additional_arguments(self, subparser) -> None:
                 default=False,
                 help="Overwrite an existing index, think twice before using.")
 
+    # Exception for invalid JSON.
+    class InvalidInputJson(Exception):
+        def __init__(self, error_message, additional_info):
+            self.error_message = error_message
+            self.additional_info = additional_info
+            super().__init__()
```

Comment:
The `Exception` class also has the method `add_note`. Here the second parameters is some the object that caused the exception to be raised. Maybe it can be renamed to `triggering_object` or `cause_object`.

Reply:
Thanks for the info, I did not know that. But I don't see a reason to use it here. And the error handling and code is already pretty verbose.